### ISSUE TITLE
topicaliases.TAHandler does not send a topic on first publish

### ISSUE THREAD

**Issue by shirou** [eclipse-paho/paho.golang#271]
**Describe the bug**

When using `topicaliases.TAHandler` to publish, the server disconnects. The ReasonCode is `0x82`, indicating a `Protocol Error`.

I believe this issue occurs because the TAHandler [does not send the topic](https://github.com/eclipse-paho/paho.golang/blob/master/paho/extensions/topicaliases/topicliases.go#L110) when the alias does not exist, It should send the topic in the first publish.

**To reproduce**

```go
	ta := topicaliases.NewTAHandler(maxTopicAlias)
	cliCfg := autopaho.ClientConfig{
		ConnectPacketBuilder: func(c *paho.Connect, u *url.URL) *paho.Connect {
			m := uint16(maxTopicAlias)
			c.Properties = &paho.ConnectProperties{
				TopicAliasMaximum: &m,
			}
			return c
		},
		ClientConfig: paho.ClientConfig{
		(snip)
			PublishHook: ta.PublishHook,
		},
	}
```

and Publish the message.

**Expected behaviour**

Server does not disconnect and publish successed.

**Software used:**
- Server : AWS IoT Core
- Client version : github.com/eclipse/paho.golang v0.21.0

**Comment by MattBrittan** [comment]
Thanks for this fix. It would be nice to see an example for using topicaliases, especially with AutoPaho (where it would be important to clear the aliases when a new conneciton is established - as per the spec "Alias mappings exist only within a Network Connection and last only for the lifetime of that Network Connection").

**Comment by shirou** [comment]
Thank you for the merge. As you mentioned, it results in an error if we don’t clear the map within the TopicAlias Handler before reconnecting. Locally, I added the following function to TAHandler and set it to be called upon `OnServerDisconnect`:

```
// ResetAlias reassigns a given alias number for a new topic
func (t *TAHandler) ResetAll() {
        t.Lock()
        defer t.Unlock()

        clear(t.aliases)
}
```
However, I found that using a Queue causes an another issue. When using topic aliases, messages stored in the Queue do not contain the topic itself. Therefore, if we attempt to send the messages saved in the Queue after reconnecting, it results in an error. I plan to open a separate issue for this.
